src/services/auth.service.ts

```diff
@@ -30,17 +33,22 @@ export class AuthorizationService {
         token = token.replace('Bearer ', '');
       }
       const payload = await jwt.verifyJWT(token);
-      const {
-        data: { email }
-      } = payload;
       const tokenIsBlacklisted: number = await redis.isMemberOfSet({
-        email,
+        email: payload.data.email,
         token
       });
-      if (!!tokenIsBlacklisted) {
-        return false;
-      }
-      return true;
+      
+      if (!!tokenIsBlacklisted) return false;
+      
+      // if there is no role restriction on the endpoint allow access
+      if (!_roles.length) return true;
+      // if not, continue checking if the user's role is in the list
+      const user = await userService.findUserByEmail(payload.data.email);
+      const roleF = _roles.find(role =>
+        !!user.role ? user.role === role : role === Role.USER
+      );
```

Comment:
What is the purpose of this? Besides the logic, "roleF" is not a significant name for a variable. A good practice is declaring things that could be easily readable and understandable by everybody on the team.

Reply:
fixed, renamed variable